src/internal/operators/repeat.ts

```diff
@@ -7,8 +7,41 @@ import { MonoTypeOperatorFunction, TeardownLogic } from '../types';
 /**
  * Returns an Observable that will resubscribe to the source stream when the source stream completes, at most count times.
  *
+ * <span class="informal">Repeats all values emitted on the source. It's like {@link retry}, but for non error cases.</span>
+ *
  * ![](repeat.png)
  *
+ * Similar to {@link retry}, this operator repeats the stream of items emitted by the source for non error cases.
+ * Repeat can be useful for creating observables that are meant to have some repeated pattern or rhythm.
+ *
+ * Note: `repeat(0)` returns an empty observable and `repeat()` will repeat forever
+ *
+ * ## Example
+ * Repeat a message stream
+ * ```ts
+ * import { of } from 'rxjs';
+ * import { repeat, delay } from 'rxjs/operators';
+ *
+ * const source = of('Repeat message');
+ * const example = source.pipe(repeat(3));
+ * example.subscribe(x => console.log(x));
+ * // Results: Repeat message Repeat message Repeat message
```

Comment:
First of all, great contribution! 

I'd like to suggest one tiny improvement. Don't want to sound picky but wouldn't it be better if each output was on its own line? Because in reality that is what is happening.

Reply:
Yes, agree. I have pushed a change to have them on its own line and reduce the second example so that it takes less lines but is still comprehensible.